File: ouweh-vdwerf/src/main/org/uva/qls/ast/Widget/Widget.java
package org.uva.qls.ast.Widget;

import org.uva.qls.ast.TreeNode;
import org.uva.qls.ast.Widget.WidgetTypes.WidgetType;

public class Widget extends TreeNode {

    private WidgetType type;

    public Widget(WidgetType type){
        this.type = type;
    }

    public boolean isCompatible(String questionType) {
        return this.type.getCompatibleTypes().contains(questionType);
    }
}

File: DogeQL/src/main/kotlin/qls/ast/node/widget/Widget.kt
package qls.ast.node.widget

import qls.ast.node.attribute.Attribute
import qls.visitor.QlsVisitor

class Widget(val type: WidgetType) : Attribute {
    override fun <T> accept(visitor: QlsVisitor<T>): T {
        return visitor.visit(this)
    }
}